File: cef-web/src/main/java/gov/epa/cef/web/service/ControlAssignmentService.java
package gov.epa.cef.web.service;

import java.util.List;

import gov.epa.cef.web.service.dto.ControlAssignmentDto;

public interface ControlAssignmentService {

    /**
     * Retrieve Control Assignment by its id
     * @param id
     * @return
     */
    ControlAssignmentDto retrieveById(Long id);

    /**
     * Retrieve Control Assignments for an emissions process
     * @param processId
     * @return
     */
    List<ControlAssignmentDto> retrieveForEmissionsProcess(Long processId);

    /**
     * Retrieve Control Assignments for an emissions unit
     * @param unitId
     * @return
     */
    List<ControlAssignmentDto> retrieveForEmissionsUnit(Long unitId);

    /**
     * Retrieve Control Assignments for a release point
     * @param pointId
     * @return
     */
    List<ControlAssignmentDto> retrieveForReleasePoint(Long pointId);

}

File: cef-web/src/main/webapp/src/app/core/services/control-assignment.service.ts
import { Injectable } from '@angular/core';
import { ControlAssignment } from 'src/app/shared/models/control-assignment';
import { Observable } from 'rxjs';
import { HttpClient } from '@angular/common/http';

@Injectable({
  providedIn: 'root'
})
export class ControlAssignmentService {

  private baseUrl = 'api/controlAssignment';  // URL to web api

  constructor(private http: HttpClient) { }

  /** GET specified control assignment from the server */
  retrieve(id: number): Observable<ControlAssignment> {
    const url = `${this.baseUrl}/${id}`;
    return this.http.get<ControlAssignment>(url);
  }

  retrieveForEmissionsProcess(processId: number): Observable<ControlAssignment[]> {
    const url = `${this.baseUrl}/process/${processId}`;
    return this.http.get<ControlAssignment[]>(url);
  }

  retrieveForEmissionsUnit(unitId: number): Observable<ControlAssignment[]> {
    const url = `${this.baseUrl}/unit/${unitId}`;
    return this.http.get<ControlAssignment[]>(url);
  }

  retrieveForReleasePoint(pointId: number): Observable<ControlAssignment[]> {
    const url = `${this.baseUrl}/releasePoint/${pointId}`;
    return this.http.get<ControlAssignment[]>(url);
  }
}
